packages/wallet/src/redux/protocols/indirect-defunding/readme.md

```diff
@@ -0,0 +1,57 @@
+# Indirect De-Funding Protocol
+
+The purpose of this protocol is handle de-funding a channel that has been indirectly funded.
+
+It covers:
+
+- Checking that a channel is closed (either finalized on chain or a conclusion proof exists)
+- Crafting a ledger update that allocates the funds to the players.
+- Waiting for a ledger response from the opponent.
+
+## State machine
+
+### Player B State machine
+
+```mermaid
+graph TD
+  St((start))-->DF{Defundable?}
+  DF --> |No| F((Failure))
+  DF -->|Yes|SC0[SendLedgerUpdate0]
+  SC0-->WFU(WaitForLedgerUpdate)
+  WFU --> |"CommitmentReceived(Accept)"|SC1[SendLedgerUpdate2]
+  SC1-->S((success))
+  WFU --> |"CommitmentReceived(Reject)"| F
+```
+
+### Player A State machine
+
+```mermaid
+graph TD
+  St((start))-->DF{Defundable?}
+  DF --> |No| F((Failure))
+  DF --> |Yes| WFU(WaitForLedgerUpdate)
+  WFU-->|"CommitmentReceived(Accept)"|SC1[SendLedgerUpdate1]
+  WFU --> |"CommitmentReceived(Reject)"| F
+  SC1-->WFFU(WaitForFinalLedgerUpdate)
+  WFFU-->|"CommitmentReceived(Accept)"|S((success))
+  WFFU-->|"CommitmentReceived(Reject)"|F
+
+```
+
+Notes:
+
+- SendLedgerUpdate0/1 are not states but indicate when the ledger update is sent.
+- A single reducer implements both the player A and B state machine.
+
+Assumptions:
+
+- We assume that Player A was responsible for the original set of ledger updates to fund the channel. Due to the current consensus algorithm if player A proposed a consensus (to fund the channel), player B must propose the next consensus (to de-fund the channel).
```

Comment:
I only recently realized that Player B has to be responsible for sending the first ledger update to defund the channel. I've updated the readme but I'll need to go through and swap most of the test cases / player checks.

Longer term this might cause us headaches. We'll need to keep track of who was responsible for the last consensus round and have the next player start that consensus. For now I'm considering that out of scope.

Reply:
The simplest thing to do is for everyone but player `0` to respond to any ledger updates, either by reseting the consensus counter or agreeing with the proposed allocation.

This always leaves player `0` (player A) able to propose a ledger update.

Reply:
Isn't Player A responsible for sending the commitment that sets allocation=proposedAllocation? Meaning after all other players have accepted the consensus Player A must send out a commitment that has `allocation`=`proposedAllocation` ?

Reply:
yes, but the rest of the players can continue to increment the consensus counter, giving player A the opportunity to change `proposedAllocation`

Reply:
Ah, I think I understand. I'm going to leave Player A as the first mover for now and we can discuss next standup if we want to either:
- Change Indirect Funding so that Player B sends another ledger update
- Change Indirect Defunding so that player B is the first mover.